lib/cgi/util.rb

```diff
@@ -42,7 +42,8 @@ def escapeHTML(string)
   #      # => "Usage: foo \"bar\" <baz>"
   def unescapeHTML(string)
     enc = string.encoding
-    if [Encoding::UTF_16BE, Encoding::UTF_16LE, Encoding::UTF_32BE, Encoding::UTF_32LE].include?(enc)
+    return string unless string.include? "&".encode(enc)
```

Comment:
This `include?` raises `Encoding::CompatibilityError` when `enc` is UTF-{16,32}.
I'm uncertain that unescaping wide char is necessary or reasonable.